src/o365/spo/commands/listitem/listitem-record-declare.ts

```diff
@@ -0,0 +1,304 @@
+import auth from "../../SpoAuth";
+import config from "../../../../config";
+import commands from "../../commands";
+import GlobalOptions from "../../../../GlobalOptions";
+import * as request from "request-promise-native";
+import {
+  CommandOption,
+  CommandValidate,
+  CommandTypes,
+  CommandError
+} from "../../../../Command";
+import SpoCommand from "../../SpoCommand";
+import Utils from "../../../../Utils";
+import { Auth } from "../../../../Auth";
+import {
+  ContextInfo,
+  ClientSvcResponse,
+  ClientSvcResponseContents,
+} from "../../spo";
+import { ClientSvc, IdentityResponse } from "../../common/ClientSvc";
+
+const vorpal: Vorpal = require("../../../../vorpal-init");
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  listId?: string;
+  listTitle?: string;
+  id: string;
+  date?: string;
+}
+
+class SpoListItemRecordDeclareCommand extends SpoCommand {
+
+  public get name(): string {
+    return commands.LISTITEM_RECORD_DECLARE;
+  }
+
+  public get description(): string {
+    return "Declares the specified list item as a record";
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.listId = typeof args.options.listId !== "undefined";
+    telemetryProps.listTitle = typeof args.options.listTitle !== "undefined";
+    telemetryProps.date = typeof args.options.date !== "undefined";
+    return telemetryProps;
+  }
+
+  public commandAction(
+    cmd: CommandInstance,
+    args: CommandArgs,
+    cb: () => void
+  ): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    const clientSvc: ClientSvc = new ClientSvc(cmd, this.debug);
+    const listIdArgument = args.options.listId || "";
+    const listTitleArgument = args.options.listTitle || "";
+    const id = args.options.id;
+    const dateArgument = args.options.date || "";
+    let siteAccessToken: string = "";
+    let formDigestValue: string = "";
+    let webIdentity: string = "";
+    let listId: string = "";
+
+
+    const listRestUrl: string = args.options.listId
+      ? `${args.options.webUrl}/_api/web/lists(guid'${encodeURIComponent(
+        listIdArgument
+      )}')`
+      : `${args.options.webUrl}/_api/web/lists/getByTitle('${encodeURIComponent(
+        listTitleArgument
+      )}')`;
+
+    if (this.debug) {
+      cmd.log(`Retrieving access token for ${resource}...`);
+    }
+
+    auth
+      .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then(
+        (accessToken: string): request.RequestPromise => {
+          siteAccessToken = accessToken;
+
+          if (this.debug) {
+            cmd.log(
+              `Retrieved access token ${accessToken}. Retrieving request digest...`
+            );
+          }
+
+          return this.getRequestDigest(cmd, this.debug);
```

Comment:
Here we should use the `getRequestDigestForSite` method instead passing the URL of the site specified in the webUrl option. Otherwise, the command will try to retrieve request digest for the site used to connect to SharePoint which will typically fail.